### PR TITLE
[G] Category bottom sheet

### PR DESCRIPTION
**Pull request by GholemHub**

### CODE REVIEW

File: app/src/main/java/com/gholem/moneylab/domain/model/TransactionCategory.kt [hunk 1]
```diff
@@ -18,7 +18,6 @@ enum class TransactionCategory(
     companion object {
 
         fun getDefault() = OTHER
-
```
Comment (bartoszrosol1) [inline app/src/main/java/com/gholem/moneylab/domain/model/TransactionCategory.kt]: There should be an empty line between methods

File: app/src/main/java/com/gholem/moneylab/features/add/adapter/AddTransactionsAdapter.kt [hunk 2]
```diff
@@ -12,8 +12,11 @@ import com.gholem.moneylab.domain.model.Transaction
 import com.gholem.moneylab.domain.model.TransactionCategory
 import com.gholem.moneylab.features.add.adapter.viewholder.AddTransactionViewHolder
 import com.gholem.moneylab.util.timestampToString
+import timber.log.Timber.d
 
-class AddTransactionsAdapter : RecyclerView.Adapter<AddTransactionViewHolder>() {
+//Listener(val categoryClickListener: () -> Unit)
```
Comment (bartoszrosol1) [inline app/src/main/java/com/gholem/moneylab/features/add/adapter/AddTransactionsAdapter.kt]: Remove commented line

File: app/src/main/java/com/gholem/moneylab/features/add/adapter/AddTransactionsAdapter.kt [hunk 3]
```diff
@@ -12,8 +12,11 @@ import com.gholem.moneylab.domain.model.Transaction
...
+class AddTransactionsAdapter(val categoryClickListener: () -> Unit) :
+    RecyclerView.Adapter<AddTransactionViewHolder>() {
```
Comment (bartoszrosol1) [inline app/src/main/java/com/gholem/moneylab/features/add/adapter/AddTransactionsAdapter.kt:20]: Format it better:

```
class AddTransactionsAdapter(
    val categoryClickListener: () -> Unit
) : RecyclerView.Adapter<AddTransactionViewHolder>() {
```

File: app/src/main/java/com/gholem/moneylab/features/add/adapter/AddTransactionsAdapter.kt [hunk 4]
```diff
@@ -72,23 +75,39 @@ class AddTransactionsAdapter : RecyclerView.Adapter<AddTransactionViewHolder>()
...
-            )
+    private fun createCategoryViewHolder(
+        parent: ViewGroup
+    ): AddTransactionViewHolder.CategoryViewHolder {
+        val binding = ItemCategoryBinding.inflate(
+            LayoutInflater.from(parent.context),
+            parent,
+            false
         )
+        //Listener
```
Comment (bartoszrosol1) [inline app/src/main/java/com/gholem/moneylab/features/add/adapter/AddTransactionsAdapter.kt]: Comments like that are unnecessary, you can clearly say that it is a click listener looking at the code

File: app/src/main/java/com/gholem/moneylab/features/add/adapter/AddTransactionsAdapter.kt [hunk 5]
```diff
@@ -102,6 +121,7 @@ class AddTransactionsAdapter : RecyclerView.Adapter<AddTransactionViewHolder>()
         return AddTransactionViewHolder.NewTransactionViewHolder(binding)
             .also { viewHolder ->
                 binding.createNewTransactionBtn.setOnClickListener {
+                    d("Start At: ${adapterData.size}")
```
Comment (bartoszrosol1) [inline app/src/main/java/com/gholem/moneylab/features/add/adapter/AddTransactionsAdapter.kt]: Is it still needed for anything? If not then remove it.

Anyway, if you are adding logs then don't import methods like `d()`, `i()`, `e()` etc. - just import `Timber` and use `Timber.i()`

File: app/src/main/java/com/gholem/moneylab/features/add/adapter/viewholder/AddTransactionViewHolder.kt [hunk 6]
```diff
@@ -19,8 +19,6 @@ sealed class AddTransactionViewHolder(binding: ViewBinding) :
             binding.categoryButton.setText(category.category.categoryName)
             binding.categoryButton
                 .setCompoundDrawablesWithIntrinsicBounds(0, 0, category.category.image, 0)
-//            binding.itemCategoryName.setText(category.category.categoryName)
-//            binding.itemCategoryImage.setBackgroundResource(category.category.image)
```
Comment (bartoszrosol1) [inline app/src/main/java/com/gholem/moneylab/features/add/adapter/viewholder/AddTransactionViewHolder.kt:23]: Remove commented lines

File: app/src/main/java/com/gholem/moneylab/main/MainActivity.kt [hunk 7]
```diff
@@ -8,6 +8,7 @@ import androidx.navigation.ui.setupWithNavController
 import com.gholem.moneylab.R
 import com.gholem.moneylab.common.BottomNavigationVisibilityBus
 import com.gholem.moneylab.databinding.ActivityMainBinding
+import com.gholem.moneylab.features.add.BottomSheetCategoryFragment
```
Comment (bartoszrosol1) [inline app/src/main/java/com/gholem/moneylab/main/MainActivity.kt]: Remove unused import

File: app/src/main/java/com/gholem/moneylab/main/MainActivity.kt [hunk 8]
```diff
@@ -27,6 +28,8 @@ class MainActivity : AppCompatActivity() {
         showBottomNavigation(
             savedInstanceState?.getBoolean(BOTTOM_NAVIGATION_VISIBILITY_KEY) ?: false
         )
+
```
Comment (bartoszrosol1) [inline app/src/main/java/com/gholem/moneylab/main/MainActivity.kt]: Remove empty lines

File: app/src/main/res/layout/item_bottom_sheet_category.xml [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toTopOf="parent"
+        tools:srcCompat="@drawable/ic_category_food" />
+
+    <TextView
+        android:id="@+id/category_name"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_marginStart="@dimen/margin_default"
+        android:text="TextView"
```
Comment (bartoszrosol1) [inline app/src/main/res/layout/item_bottom_sheet_category.xml]: Remove text assignment

File: app/src/main/res/navigation/add_navigation.xml [hunk 10]
```diff
@@ -9,5 +9,18 @@
...
+            app:destination="@+id/category_bottom_sheet_dialog"/>
+    </fragment>
+
+    <dialog
+        android:id="@+id/category_bottom_sheet_dialog"
+        android:name="com.gholem.moneylab.features.add.BottomSheetCategoryFragment"
+        android:label="CategoryBottomSheet"
+        tools:layout="@layout/bottomsheet_category_fragment"
+        
+        />
```
Comment (bartoszrosol1) [inline app/src/main/res/navigation/add_navigation.xml]: Move `/>` to the end of the last line of the dialog

**Review by bartoszrosol1 — APPROVED** [review]

File: app/src/main/java/com/gholem/moneylab/features/add/adapter/AddTransactionsAdapter.kt (hunk 2 above)
Comment (GholemHub) [reply app/src/main/java/com/gholem/moneylab/features/add/adapter/AddTransactionsAdapter.kt]: it was the comment for myself, i change it, but don't erace

**Pull request merged**